### ISSUE TITLE
HA upgrades with multiple app instances have noticeable app downtime

### ISSUE THREAD

**Issue by gaktive** [cloudfoundry-incubator/kubecf#1257]
As spotted during kubecf 2.2.2 to 2.2.3 upgrade testing, we're getting reports that app downtime is obvious. This is despite having an HA setup with multiple app instances and load distributed across Diego cells.

@HartS and @svollath can provide details on their environments but it's across different Kubernetes implementations.

**Comment by mook-as** [comment]
As far as I can tell, the HA configuration in catapult in fact doesn't HA the database (because it doesn't respect the HA flag correctly).

Still investigating.

**Comment by mook-as** [comment]
The actual reason seems to be because the `router-public` service has:
```yaml
selector:
  app.kubernetes.io/component: router
  app.kubernetes.io/instance: susecf-scf
  app.kubernetes.io/version: v2.2.3
```
The service has the version as part of the selector, so it will never match the old router (only the new one once it rolls over).  That's defined [here](https://github.com/cloudfoundry-incubator/kubecf/blob/c4af91cf8a779405b41a33ad44dacc3d800003ea/chart/templates/ingress.yaml#L121).

Judging by the history, we've never had it work before? That code's been in there since v0.1.0.

**Comment by jandubois** [comment]
> Judging by the history, we've never had it work before? That code's been in there since v0.1.0.

Yeah, but what about SCF? 😉

**Comment by mook-as** [comment]
Fixed in #1325 — the upgrades are still likely to _fail_ (because of other issues), but the apps stay up at least? :D
